src/main/java/me/dinowernli/grpc/polyglot/command/ServiceList.java

```diff
@@ -20,97 +22,146 @@
 
   /** Lists the GRPC services - filtered by service name (contains) or method name (contains) */
   public static void listServices(
-      Output output,
-      FileDescriptorSet fileDescriptorSet,
-      String protoDiscoveryRoot,
-      Optional<String> serviceFilter,
-      Optional<String> methodFilter,
-      Optional<Boolean> withMessage) {
+    Output output,
+    FileDescriptorSet fileDescriptorSet,
+    String protoDiscoveryRoot,
+    Optional<String> serviceFilter,
+    Optional<String> methodFilter,
+    Optional<Boolean> withMessage,
+    Optional<Boolean> jsonOutput) {
+
+    ArrayList<Service> services = new ArrayList<>(
+      getServices(fileDescriptorSet, protoDiscoveryRoot, serviceFilter, methodFilter, withMessage, jsonOutput));
+
+    if (jsonOutput.isPresent() && jsonOutput.get()) {
+      Gson gson = new Gson();
+      output.writeLine(gson.toJson(services));
```

Comment:
I think that's what you get when you print a message descriptor. What you actually want is the service descriptor [1]. That should get you the service name, the method name, and the type name for the messages. Service descriptors are already conveniently returns by listServices()

[1] https://github.com/google/protobuf/blob/master/src/google/protobuf/descriptor.proto#L230